### DISCUSSION TITLE
Multiple commands to trigger lazy loading (znap function)

### DISCUSSION THREAD

**Discussion by michaeldebetaz** [marlonrichert/zsh-snap#279] [Q&A]
Hi everyone 👋

I am very happy with `znap` and, as many, I use it with the nvm plugin. As nvm takes forever to load, I leverage the very useful `znap function nvm load_nvm()` trick that saves me time when loading zsh.

But I sometimes need to trigger the lazy loading with other commands, like `npm` `npx` or `nvim`.

Here is what I do in my .zshrc file to ensure that it triggers for each command:
```sh
load_nvm() {
    export NVM_DIR="$HOME/.nvm"
    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
    [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
}
znap function nvm load_nvm
znap function npm load_nvm
znap function node load_nvm
znap function npx load_nvm
znap function nvim load_nvm
```

Is this the correct way to do so or am I doing something wrong?

**Comment by marlonrichert** [comment]
[As mentioned in the Readme](https://github.com/marlonrichert/zsh-snap/blob/main/README.md#zshrc-optimization), `znap function` can create multiple lazy-loading functions at once. Just pass all the command names you want to lazy-load before passing the actual code to run. So, for your case:
```sh
load_nvm() {
    export NVM_DIR="$HOME/.nvm"
    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
    [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
}
znap function nvm npm node npx nvim load_nvm
```
Znap will auto-remove all five of the functions this create when you first try to run any of these five commands.

**Reply by michaeldebetaz** [reply]
Thank you for the clarification! I read the part of the Readme, but it wasn't clear for me that this was the implementation. Sorry for my lack of understanding 😋

Again, thank you so much for your work!
